### PR TITLE
Triangulation_3: Add a function to file_input() with a similar API to the advanced copy_tds()

### LINKED ISSUE #946: Triangulation_file_input.h, similar to copy_TDS
On one hand, there is that accepted small feature: https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/More_general_copy_tds

On the other hand, there is the file [`Polyhedron/demo/Polyhedron/include/CGAL/Triangulation_file_input.h`](https://github.com/CGAL/cgal/blob/releases/CGAL-4.8-beta2/Polyhedron/demo/Polyhedron/include/CGAL/Triangulation_file_input.h) that uses a similar method, with different names.

I suggest that we unify the API of the two, and submit the improved `Triangulation_file_input.h`.

To: @sloriot @janetournois 
Cc: @afabri

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Move the file Triangulation_file_input.h form the demo to the Triangulation_3 package, rework the API to match the one from `TDS_3::copy_tds()` that uses functors, write some doc and adapt the impacted plugins of the demo.

## Release Management

* Affected package(s): Triangulation_3
* Issue(s) solved (if any): fix #946 
* Feature/Small Feature (if any): [Small Feature](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Add_function_file_input_to_T3#Observations_and_Discussion)
* **pre-approuved by @lrineau and @MoniqueTeillaud, 2020/02/28**
* Link to compiled documentation (obligatory for small feature) See in the SF
* License and copyright ownership:
* CHANGES.md : 
```
### 3D Triangulations
-   The free function `CGAL::file_input()` and the member function `CGAL::Triangulation_3::file_input()` 
    have been added. The first allows to load a `Triangulation_3` from an input stream, using functors to create vertices and cells.
    The second is simply the member function version of the first one. 


### 3D Triangulation Data Structure
-   The free function `CGAL::file_input()` and the member function `CGAL::TDS_3::file_input()` 
    have been added. The first allows to load a `TDS_3` from an input stream, using functors to create vertices and cells.
    The second is simply the member function version of the first one. 
```

TODO:
 - [x] add a member function to Triangualtion_3 that calls file_input().
 - [x] write a small feature
 - [x] add a test
 - [x] Write the `CHANGES.md`

### CODE REVIEW

File: Triangulation_3/include/CGAL/IO/Triangulation_file_input.h [hunk 1]
```diff
@@ -88,10 +87,9 @@ std::istream& file_input(std::istream& is, Tr2 &tr,
   for (std::size_t j=0 ; j < m; j++) {
     Cell1 c;
     if(!(is >> c)) return is;
-    if(!update_cell(c, *(C[j]))) {
-      is.setstate(std::ios_base::failbit);
-      return is;
-    }
+    Cell_handle ch=tr.tds().create_cell(convert_cell(c));
```
Comment (lrineau) [inline Triangulation_3/include/CGAL/IO/Triangulation_file_input.h]: Have you tested? I am pretty sure that is wrong: cells are already created in line 85:
```
 tr.tds().read_cells(is, V, m, C);
```

Comment (maxGimeno) [reply Triangulation_3/include/CGAL/IO/Triangulation_file_input.h]: It works in the demo :(

Comment (afabri) [reply Triangulation_3/include/CGAL/IO/Triangulation_file_input.h]: Not really a proof.   You might create cells that are not needed and not interconnected with other cells.

Comment (maxGimeno) [reply Triangulation_3/include/CGAL/IO/Triangulation_file_input.h]: Yes, I think there were simply too many cels in the created c3t3 and I did not see.

**Comment by maxGimeno** [comment]
Before doing the SF, @lrineau wanted me to ask for pre-reviews for the documentation of those I/O functions. Because we do not have a standard format for triangulation_3, I had to make some  odd explainations about how to obtain the input, and it really needs to be reviewed. @CGAL/geometryfactory

**Comment by maxGimeno** [comment]
ping @sloriot @afabri @MaelRL 
Can one of you review the doc ?

**Comment by MoniqueTeillaud** [comment]
Note that following our rules, I am supposed to review this submission. I have started a discussion in the wiki page of this small feature

**Comment by maxGimeno** [comment]
Success in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-119.shtml

**Comment by sloriot** [comment]
@lrineau is a code review required or have you already done it ?

**Comment by lrineau** [comment]
I remember that we discussed quite a bit about the code, while Maxime was working on this PR. I think it is good.

**Pull request merged**